src/search/command_line.cc

```diff
@@ -184,6 +185,147 @@ shared_ptr<SearchAlgorithm> parse_cmd_line(
     return parse_cmd_line_aux(args);
 }
 
+static void complete_filename(const string &prefix, vector<string> &suggestions) {
+    // Split into directory and file_prefix
+    filesystem::path prefix_path(prefix);
+    if (!filesystem::is_directory(prefix_path)) {
+        prefix_path = prefix_path.parent_path();
+    }
+
+    vector<filesystem::path> path_suggestions;
+    if (prefix_path.empty()) {
+        for (const auto &entry : filesystem::directory_iterator(filesystem::current_path())) {
+            path_suggestions.push_back(entry.path().filename());
+        }
+    } else if (filesystem::is_directory(prefix_path)) {
+        for (const auto &entry : filesystem::directory_iterator(prefix_path)) {
+            path_suggestions.push_back(entry.path());
+        }
+    }
+
+    for (const filesystem::path &path : path_suggestions) {
+        string suggestion = path.string();
+        if (filesystem::is_directory(path)) {
+            /*
+              Append preferred separator ("/" or "\") to directories based on
+              operating system.
+            */
+            suggestion += filesystem::path::preferred_separator;
+        }
+        suggestions.push_back(suggestion);
+    }
+}
+
+static vector<string> complete_args(
+    const vector<string> &parsed_args, const string &current_word,
+    int cursor_pos) {
+    string prefix = current_word.substr(0, cursor_pos);
+    assert(!parsed_args.empty()); // args[0] is always the program name.
+    const string &last_arg = parsed_args.back();
+    vector<string> suggestions;
+    if (find(parsed_args.begin(), parsed_args.end(), "--help") != parsed_args.end()) {
+        suggestions.push_back("--txt2tags");
+        plugins::Registry registry = plugins::RawRegistry::instance()->construct_registry();
+        for (const shared_ptr<const plugins::Feature> &feature : registry.get_features()) {
+            suggestions.push_back(feature->get_key());
+        }
+    } else if (last_arg == "--internal-plan-file") {
+        // Suggest filename starting with current_word
+        complete_filename(prefix, suggestions);
+    } else if (last_arg == "--internal-previous-portfolio-plans") {
+        // no suggestions, integer expected
+    } else if (last_arg == "--search") {
+        // suggestions in search string based on current_word
+        plugins::Registry registry = plugins::RawRegistry::instance()->construct_registry();
+        for (const shared_ptr<const plugins::Feature> &feature : registry.get_features()) {
+            suggestions.push_back(feature->get_key() + "(");
```

Comment:
For me this shows on the command line as "adapt_costs\(", and if I use the autocomplete I also get the backslash.
Furthermore, this only works for the "outermost" part, i.e. if I already entered "astar(" then it won't suggest anything anymore (but I presmue you know this).

Reply:
This is a very naive way of adding completions. For now, we will leave the `--search` argument mostly as it is. It will be more complicated to get this working.

Reply:
We completely disabled this now since it is not really helpful.